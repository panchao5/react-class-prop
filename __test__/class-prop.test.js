import { create } from "react-test-renderer";

describe("class prop", () => {
  test("accepts arrays", () => {
    const tree = create(<div class={["a", "b"]}>Hello World</div>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("accepts nested arrays", () => {
    const tree = create(<div class={["a", "b"]}><span class={["c", ["d"]]}>Yo</span></div>).toJSON();

    expect(tree).toMatchSnapshot(); 
  })

  test("accepts objects", () => {
    const tree = create(
      <div class={{ a: false, b: true }}>Hello World</div>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("appends to className", () => {
    const tree = create(
      <div class={["b", "c"]} className="a">
        Hello World
      </div>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
