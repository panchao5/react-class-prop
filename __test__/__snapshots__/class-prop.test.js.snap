// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class prop accepts arrays 1`] = `
<div
  className="a b"
>
  Hello World
</div>
`;

exports[`class prop accepts nested arrays 1`] = `
<div
  className="a b"
>
  <span
    className="c d"
  >
    Yo
  </span>
</div>
`;

exports[`class prop accepts objects 1`] = `
<div
  className="b"
>
  Hello World
</div>
`;

exports[`class prop appends to className 1`] = `
<div
  className="a b c"
>
  Hello World
</div>
`;
